apm-sniffer/bootstrap-plugins/jdk-threadpool-plugin/src/main/java/org/apache/skywalking/apm/plugin/AbstractThreadingPoolInterceptor.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin;
+
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+
+import java.lang.reflect.Method;
+
+public abstract class AbstractThreadingPoolInterceptor implements InstanceMethodsAroundInterceptor {
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes, MethodInterceptResult result) throws Throwable {
+        if (!ContextManager.isActive()) {
+            return;
+        }
+
+        if (allArguments == null || allArguments.length != 1) {
+            return;
+        }
+        Object argument = allArguments[0];
+
+        Object updateResult = updateOnlyParam(argument);
+        if (updateResult != null) {
+            allArguments[0] = updateResult;
+        }
+    }
+
+    public abstract Object updateOnlyParam(Object param);
```

Comment:
What does this method name mean? `onlyParam` means?

Reply:
> 

it will change the param of `execute` and `submit` method wrapping it to `SWCallableWrapper` or `SWRunnableWrapper`.

Reply:
Then we should name it `wrapExecutor`?

Reply:
> 

it doesn't wrap executor. it just wraps the `Runnable` or `Callable`.  we could name it `wrapCommand` or `wrapTask`. WDYT?

Reply:
Just `wrap` should be enough, I think.

Reply:
> Just `wrap` should be enough, I think.

ok